node/service/src/grandpa_support.rs

```diff
@@ -155,7 +155,7 @@ impl<B> grandpa::VotingRule<PolkadotBlock, B> for ApprovalCheckingVotingRule
 				let diff = best_number.saturating_sub(base_number);
 				if diff >= MAX_APPROVAL_CHECKING_FINALITY_LAG {
 					// Catch up to the best, with some extra lag.
-					let target_number = best_number - MAX_APPROVAL_CHECKING_FINALITY_LAG;
+					let target_number = best_number.saturating_sub(MAX_APPROVAL_CHECKING_FINALITY_LAG);
```

Comment:
I see this, but maybe a comment to the effect of

`diff >= MAX_LAG` & `diff = best - base`, therefore `best >= MAX_LAG + base`, therefore `best >= MAX_LAG`?

Reply:
Simplified with `checked_sub` which returns `None` on underflow. Assumption is that `BlockNumber` is sufficiently large to never overflow when adding `MAX_APPRO.._LAG`.